network/endpoint_windows.go

```diff
@@ -44,10 +40,69 @@ func ConstructEndpointID(containerID string, netNsPath string, ifName string) (s
 
 // newEndpointImpl creates a new endpoint in the network.
 func (nw *network) newEndpointImpl(epInfo *EndpointInfo) (*endpoint, error) {
+	// Check for missing namespace
+	if epInfo.NetNsPath == "" {
+		log.Printf("[net] Endpoint missing Namsepace, cannot Create. [%v].", epInfo)
+		return nil, fmt.Errorf("Cannot create Endpoint without a Namespace")
+	}
+
 	// Get Infrastructure containerID. Handle ADD calls for workload container.
 	var err error
 	infraEpName, _ := ConstructEndpointID(epInfo.ContainerID, epInfo.NetNsPath, epInfo.IfName)
 
+	/*
+		// TODO: RS5 at release does not process V2 Endpoint Schema, using V1 for now.
+		hcnPolicies := policy.SerializeHostComputeEndpointPolicies(epInfo.Policies)
+
+		hnsEndpoint := &hcn.HostComputeEndpoint{
```

Comment:
HostComputeEndpoint is the endpoint added to the particular network which already existed or it was created on the node right?

Reply:
Correct, the network must already exist before calling this method. The name 'HNSEndpoint' was updated to 'HostComputeEndpoint' in the Schema update, but they are fundamentally the same.

Reply:
thanks for explanation